jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/archive/AssemblyFileSetUtils.java

```diff
@@ -169,8 +169,10 @@ private static List<AssemblyFileEntry> copy(Path sourceDirectory, File source, F
         for (File sourceChild : Optional.ofNullable(source.listFiles()).orElse(new File[0])) {
           copy(sourceDirectory, sourceChild, new File(target, sourceChild.getName()), assemblyFileSet);
         }
+        FileUtil.updateLastModifiedTimestamp(target, source.lastModified());
```

Comment:
I think it'd be better if this was done as part of the copy.
Instead of adding an extra-function, just modify FileUtil.copy to preserve the original metadata

Reply:
OK, in this case target refers to the directoy.
Likewise, whatever process was used to create the directory should be used to preserve that timestamp.
I don't see the logic (besides for this current use-case) in updating the timestamp of an existent directory.

Reply:
Umm, you mean FileUtil.copy?

I noticed that whenever a new file gets added to directory, it's lastModified timestamp gets updated. This is done to revert directory's lastModified to original source's lastModified